src/Processors/Formats/Impl/IonRowOutputFormat.cpp

```diff
@@ -0,0 +1,323 @@
+            writer->writeNull();
+            return;
+        }
+        case TypeIndex::Map: {
+            const auto & map_column = assert_cast<const ColumnMap &>(column);
+            const auto & nested_column = map_column.getNestedColumn();
+            const auto & key_value_columns = map_column.getNestedData().getColumns();
+            const auto & key_column = key_value_columns[0];
+            const auto & value_column = key_value_columns[1];
+
+            const auto & map_type = assert_cast<const DataTypeMap &>(*data_type);
+            const auto & offsets = nested_column.getOffsets();
+            size_t offset = offsets[row_num - 1];
+            size_t size = offsets[row_num] - offset;
+            writer->writeStartStruct();
+            for (size_t i = 0; i < size; ++i)
+            {
+                if (map_type.getKeyType()->getTypeId() == TypeIndex::String)
+                    writer->writeStructFieldName(assert_cast<const ColumnString &>(*key_column).getDataAt(offset + i).toView());
+                else if (map_type.getKeyType()->getTypeId() == TypeIndex::FixedString)
+                    writer->writeStructFieldName(assert_cast<const ColumnFixedString &>(*key_column).getDataAt(offset + i).toView());
+                else
+                    throw Exception(ErrorCodes::BAD_ARGUMENTS, "Ion only supports string types for map keys");
+                serializeField(*value_column, map_type.getValueType(), offset + i);
+            }
+            writer->writeFinishStruct();
+            return;
+        }
+        case TypeIndex::LowCardinality: {
+            const auto & lc_column = assert_cast<const ColumnLowCardinality &>(column);
+            auto dict_type = assert_cast<const DataTypeLowCardinality *>(data_type.get())->getDictionaryType();
+            auto dict_column = lc_column.getDictionary().getNestedColumn();
+            size_t index = lc_column.getIndexAt(row_num);
+            serializeField(*dict_column, dict_type, index);
+            return;
+        }
+        case TypeIndex::UUID: {
+            const auto & uuid_column = assert_cast<const ColumnUUID &>(column);
+            WriteBufferFromOwnString buf;
+            writeText(uuid_column.getElement(row_num), buf);
+            writer->writeString(buf.stringView());
+            return;
+        }
+        default:
+            break;
+    }
+    throw Exception(ErrorCodes::ILLEGAL_COLUMN, "Type {} is not supported for MsgPack output format", data_type->getName());
+}
+
+void IonRowOutputFormat::writeRowStartDelimiter()
+{
+    writer->writeStartStruct();
+}
+
+void IonRowOutputFormat::writeRowEndDelimiter()
+{
+    writer->writeFinishStruct();
+}
+
+void IonRowOutputFormat::writePrefix()
+{
+    writer->writeStartList();
+}
+
+void IonRowOutputFormat::writeSuffix()
+{
+    writer->writeFinishList();
+}
+
+void IonRowOutputFormat::finalizeImpl()
+{
+    writer->final();
+}
+
+void IonRowOutputFormat::write(const Columns & columns, size_t row_num)
```

Comment:
Why do we need to override `IRowOutputFormat::write` method when we can override `IRowOutputFormat::writeField`? `IRowOutputFormat::write` default implementation will call `writeRowStartDelimiter/writeField/writeRowEndDelimiter`

Reply:
Still relevant